apps/egghead/src/inngest/functions/egghead/create-course-in-egghead.ts

```
@@ -0,0 +1,87 @@
+import { db } from '@/db'
+import { contentResource } from '@/db/schema'
+import { EGGHEAD_COURSE_CREATED_EVENT } from '@/inngest/events/egghead/course-created'
+import { POST_CREATED_EVENT } from '@/inngest/events/post-created'
+import { inngest } from '@/inngest/inngest.server'
+import { createEggheadCourse } from '@/lib/egghead'
+import { loadEggheadInstructorForUser } from '@/lib/users'
+import { eq } from 'drizzle-orm'
+import { NonRetriableError } from 'inngest'
+
+export const createCourseInEgghead = inngest.createFunction(
+	{
+		id: 'create-course-in-egghead',
+		name: 'Create Course in Egghead',
+	},
+	{
+		event: POST_CREATED_EVENT,
+	},
+	async ({ event, step }) => {
+		const post = await step.run('verify-post', async () => {
+			const { post } = event.data
+			if (!post) {
+				throw new NonRetriableError('Post not found')
+			}
+
+			if (post.fields?.postType !== 'course') {
+				throw new NonRetriableError('Post is not a course')
+			}
+
+			return post
+		})
+
+		const instructor = await step.run('get-instructor', async () => {
+			const instructor = await loadEggheadInstructorForUser(post.createdById)
+
+			if (!instructor) {
+				return null
```

this could throw the error instead of returning null and probably remove the need for the guard below